prExplorer/client-app/src/app/stores/productStore.ts

```diff
@@ -0,0 +1,115 @@
+import { makeAutoObservable, runInAction } from "mobx";
+import { Product } from "../models/product";
+import agent from "../api/agent";
+import { v4 as uuid } from "uuid";
+
+export default class ProductStore {
+  productRegistry = new Map<string, Product>();
+  selectedProduct: Product | undefined = undefined;
+  editMode = false;
+  loading = false;
+  loadingInitial = true;
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  get productsSortedByName() {
+    const productList = Array.from(this.productRegistry.values());
+    productList.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+      return titleA.localeCompare(titleB);
+    });
+    return productList;
+  }
+
+
+  loadProducts = async () => {
+    try {
+      const products = await agent.Products.list();
+      products.forEach((product) => {
+        this.productRegistry.set(product.id, product);
+      });
+      this.setLoadingInitial(false);
+    } catch (error) {
+      console.log(error);
+      this.setLoadingInitial(false);
+    }
+  };
+
+  setLoadingInitial = (state: boolean) => {
+    this.loadingInitial = state;
+  };
+
+  selectProduct = (id: string) => {
+    this.selectedProduct = this.productRegistry.get(id);
+  };
+
+  cancelSelectedProduct = () => {
+    this.selectedProduct = undefined;
+  };
+
+  openForm = (id?: string) => {
```

Comment:
The `openForm` is not the best name in that case. It suggests that calling this method would somehow magically open the edit form of a product. Instead, it does two things:
1. Selects given product
2. Changes mode to edit mode

When clicking an edit button, your product should already be selected, right?
So I propose to have here an action like that:

```ts
setViewMode = (isEditMode: boolean) => {
    this.editMode = isEditMode;
  };
```

and use it accordingly from your components